### PR TITLE
Fix `internal/log/retry` pacakge

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I fixed `internal/log/retry` package based on the generated test code.
this code's coverage is 100%.

### Environment:

- Go Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/log/retry/retry_test.go [hunk 1]
```diff
@@ -25,490 +25,352 @@ import (
...
-				name: "returns nothing when fn is nil",
-				checkFunc: func() error {
-					return nil
-				},
-			}
-		}(),
+		{
+			name: "returns nothing when fn is nil",
+			checkFunc: func() error {
+				return nil
```
Comment (kevindiu) [inline internal/log/retry/retry_test.go:155]: it doesn't check anything?

Comment (hlts2) [reply internal/log/retry/retry_test.go]: thank you for your review.
This test case is not nessesary. I fixed the early return in the implementation code, so I removed this case.

Comment (kevindiu) [reply internal/log/retry/retry_test.go]: ok!

**Pull request merged**